### PR TITLE
core: add infra states and allow returning InfraCacheStatus for a given infra

### LINKED ISSUE #4007: core: add new infra states and allow the infra cache status of one infra to be returned
# Description

update post /infra_state endpoint in core to get the current state of the loading of one infra and add infra states

# Goal

We want to return the current state of an infra

# Acceptance Criteria

- [ ] The endpoint return the state of the infra

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #4007

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/api/InfraCacheStatusEndpoint.java [hunk 1]
```diff
@@ -9,13 +9,20 @@
...
 import org.takes.rs.RsJson;
 import org.takes.rs.RsWithBody;
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
 public final class InfraCacheStatusEndpoint implements Take {
     private final InfraManager infraManager;
+
+    public static final JsonAdapter<InfraCacheRequest> adapterRequest = new Moshi
```
Comment (Khoyo) [inline core/src/main/java/fr/sncf/osrd/api/InfraCacheStatusEndpoint.java:22]: I belive this should be private, if only for consistency sake

File: core/src/main/java/fr/sncf/osrd/api/InfraManager.java [hunk 2]
```diff
@@ -58,6 +58,9 @@ public enum InfraStatus {
         DOWNLOADING(false),
         PARSING_JSON(false),
         PARSING_INFRA(false),
+        KOTLIN_ADAPTATION(false),
```
Comment (Khoyo) [inline core/src/main/java/fr/sncf/osrd/api/InfraManager.java]: `ADAPTING_KOTLIN` (again, for consistency)

**Review by Khoyo — COMMENTED** [review]
Otherwise LGTM

Maybe it deserves at least a test?

**Comment by Khoyo** [comment]
(Btw, the referenced issue doesn't seem to match this pr)

File: core/src/main/java/fr/sncf/osrd/api/InfraCacheStatusEndpoint.java (hunk 1 above)
Comment (younesschrifi) [reply core/src/main/java/fr/sncf/osrd/api/InfraCacheStatusEndpoint.java:22]: As I added a test, the function must be public now

**Review by Khoyo — APPROVED** [review]

**Pull request merged**